torch/csrc/jit/passes/peephole.cpp

```diff
@@ -20,6 +20,36 @@ void PeepholeOptimize(std::shared_ptr<Graph>& graph) {
         continue;
       }
     }
+
+    // Nop transpose
+    if (n->kind() == kTranspose) {
+      bool nop = true;
+      for (size_t i = 0; i < n->is(kperm).size(); i++) {
+        if (i != n->is(kperm)[i]) {
+          nop = false;
+          break;
+        }
+      }
+      if (nop) {
+        n->replaceAllUsesWith(n->input());
+        it.destroyCurrent();
+        continue;
+      }
+    }
+
+    std::vector<int64_t> simpleTransPerm({1,0});
+
+    // Inverse transposes
+    if (n->kind() == kTranspose &&
+        n->input()->kind() == kTranspose) {
+      if (n->is(kperm) == simpleTransPerm &&
+          n->input()->is(kperm) == simpleTransPerm) {
```

Comment:
You can generalize this, right? Transposes in ONNX are secretly permutations, and permutations are closed under composition.

Reply:
yeah I was thinking this too. I'll take a stab at it.